main.rb

```diff
@@ -0,0 +1,85 @@
+module Enumerable
+  def my_each()
+    for i in self
+      yield(i)
+    end
+  end
+
+  def my_each_with_index()
+    for i in (0...self.length)
+      yield(self[i],i)
+    end
+  end
+
+  def my_select()
+    new_array =[]
+    self.my_each do |item|
+      new_array << item if yield(item) 
+    end
+    new_array
+  end
+
+  
+  def my_all?()
+    self.each do |i|
+     res = yield(i)
+     if res != true
+        return false
+      end
+    end
+    true
+  end
+
+  def my_any?()
+    self.my_each do |i|
+      res = yield(i)
+      if res == true
+        return true
+      end
+    end
+    false
+  end
+
+  def my_none?()
+    new_array = self.my_select { |i| yield(i) }
```

Comment:
As we discussed please do not reuse the `my_select ` method. My none is actually reverse of my all so break it and return the false as soon as you hit true value, with my_select you will reach the end of the array which in case of [1,2,3,4,5,6].my_none{ |x| x.even? }, should break immediately. Also if my each is producing wrong output, your my none will produce wrong as well.

Reply:
Thanks for the review @zlayabekrija